.github/workflows/pipeline-deploy-app-preprod-prod.yaml

```diff
@@ -0,0 +1,125 @@
+name: Pipeline Deployment Application production & pre-production Environment
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: "Select the Tag to deploy artefact from"
+        required: true
+        type: string
+      environment:
+        description: "Select the deployment environment"
+        required: true
+        type: string
+        options:
+          - preprod
+          - prod
+jobs:
+  metadata:
+    name: "Get Metadata"
+    uses: NHSDigital/uec-dos-management/.github/workflows/metadata.yaml@latest
+
+  derive-artefact-address:
+    name: "Get artefact location"
+    uses: NHSDigital/uec-dos-management/.github/workflows/derive-artefact-address.yaml@latest
+    with:
+      tag: ${{ inputs.tag }}
+
+  quality-check:
+    name: "Quality Check"
+    uses: NHSDigital/uec-dos-management/.github/workflows/quality-checks.yaml@latest
+    needs:
+      [
+        metadata,
+      ]
+    with:
+      environment: ${{ inputs.environment }}
+      workspace: default
+      stacks: '["application", "data_migration", "domain-test"]'
+    secrets: inherit
+
+  deploy-application-infrastructure:
+      needs:
+        [
+          quality-check,
+          metadata,
+        ]
+      uses: NHSDigital/uec-dos-management/.github/workflows/deploy-infrastructure.yaml@latest
+      with:
+        environment: ${{ inputs.environment }}
+        domain: "uec-dos-service-management"
+        workspace: default
+        stacks: '["application", "data_migration", "domain-test"]'
```

Comment:
Not sure what domain-test is but it sounds 'test-y' - do we plan to deploy that to prod or preprod ?

Reply:
I used domain-test to grab some test evidence but I think makes sense to remove it